### PR TITLE
Allow retry count and backoff to be customizable #22458

### LINKED ISSUE #22458: Allow retry count and backoff to be customizable
##### ISSUE TYPE

 - Feature Idea

##### COMPONENT NAME

cloudformation

##### SUMMARY
I have run into issues with throttling and would like to customize my retry params (number of retries and backoff) to help stabilize my application.  I am running into issues specifically with the cloudformation module and would love to be able to override the default values.

[AWSRetry](https://github.com/ansible/ansible/blob/4554e8d7694f27b9611314bd67c2f0bc1749e6e3/lib/ansible/module_utils/ec2.py#L71)
[Cloudformation](https://github.com/ansible/ansible/blob/4554e8d7694f27b9611314bd67c2f0bc1749e6e3/lib/ansible/modules/cloud/amazon/cloudformation.py#L364)

### PR DESCRIPTION
**Pull request by snowsky**
##### SUMMARY

Customize tries and delay for cloudformation module
Fixes #22458

##### ISSUE TYPE

 - Feature Pull Request

##### ANSIBLE VERSION

```

```

##### ADDITIONAL INFORMATION

This would allow pass customized values for tries and delay of backoff's parameters.

```

```

### CODE REVIEW

**Comment by tedder** [comment]
It looks like the best param names are "repeat" and "delay". I wouldn't have chosen those, but for consistency, can you update? Make sure to change the variable names too. Let me know if you saw examples that use other param names; if that's the case we can get an official recommendation.

https://github.com/ansible/ansible/blob/e5b990a55a77371439c373071c22d7d70e95a534/lib/ansible/modules/cloud/amazon/ecs_service.py#L73
https://github.com/ansible/ansible/blob/f38ca6c92f0ee06d52c3db6135d23b4c1de308d8/lib/ansible/modules/cloud/amazon/ecs_cluster.py#L43

so `needs_revision`

**Comment by snowsky** [comment]
Yes, tries variable is being used internally. Let me update the codes.

**Comment by tedder** [comment]
Need to add the default values to the docs.

**Review by ryansb — APPROVED** [review]

**Comment by snowsky** [comment]
Is this PR ready for merge?

**Comment by tedder** [comment]
yeah, `shipit`.

**Comment by snowsky** [comment]
seems need someone who has write access to merge the change.

**Comment by tedder** [comment]
Right, [that was for Ansibot to know it is approved](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md). It'll get merged eventually.

**Comment by snowsky** [comment]
got it. ansibot, listen to me, shipit. lol

**Comment by snowsky** [comment]
chatops on github +1

**Comment by jctanner** [comment]
@snowsky only maintainers can use the ship it command.

**Comment by snowsky** [comment]
@jctanner thanks, got it.

File: lib/ansible/modules/cloud/amazon/cloudformation.py [hunk 1]
```diff
@@ -99,6 +99,18 @@
     required: false
     default: null
     version_added: "2.3"
+  delay:
+    description:
+    - Number of seconds to wait
```
Comment (ryansb) [inline lib/ansible/modules/cloud/amazon/cloudformation.py]: To wait for what? Please expand these docs.

File: lib/ansible/modules/cloud/amazon/cloudformation.py [hunk 2]
```diff
@@ -99,6 +99,18 @@
...
+    required: false
+    default: 5
+    version_added: "2.3"
+  repeat:
+    description:
+    - The number of times to retry get_stack_facts operation
```
Comment (ryansb) [inline lib/ansible/modules/cloud/amazon/cloudformation.py]: Can you add more context to these docs about why a user would want to do this? What problems might they encounter that would need this value adjusted?

**Review by ryansb — CHANGES_REQUESTED** [review]

File: lib/ansible/modules/cloud/amazon/cloudformation.py (hunk 1 above)
Comment (snowsky) [reply lib/ansible/modules/cloud/amazon/cloudformation.py]: I found something wrong. Will submit the other PR.

**Comment by snowsky** [comment]
I have just committed the other change to address the changes for create/update/delete operations after the introduction of repeat and delay parameters. We don't have test cases for cloudformation. This worries me. How could we write test cases for cloudformation? Any idea?

**Comment by snowsky** [comment]
Sorry that the first commit didn't really work. The PR has been verified, please @ryansb review.

**Comment by Akasurde** [comment]
superseded by #49086

**Pull request closed without merging**